builtin/credential/github/path_config.go

```diff
@@ -23,29 +24,67 @@ func pathConfig() *framework.Path {
 are running GitHub Enterprise or an
 API-compatible authentication server.`,
 			},
+			"ttl": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: `Duration after which authentication will be expired`,
+			},
+			"max_ttl": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: `Maximum duration after which authentication will be expired`,
+			},
 		},
 
 		Callbacks: map[logical.Operation]framework.OperationFunc{
-			logical.WriteOperation: pathConfigWrite,
+			logical.WriteOperation: b.pathConfigWrite,
 		},
 	}
 }
 
-func pathConfigWrite(
+func (b *backend) pathConfigWrite(
 	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
-	conf := config{
-		Org: data.Get("organization").(string),
-	}
+	organization := data.Get("organization").(string)
 	baseURL := data.Get("base_url").(string)
 	if len(baseURL) != 0 {
 		_, err := url.Parse(baseURL)
 		if err != nil {
 			return logical.ErrorResponse(fmt.Sprintf("Error parsing given base_url: %s", err)), nil
 		}
-		conf.BaseURL = baseURL
 	}
 
-	entry, err := logical.StorageEntryJSON("config", conf)
+	var ttl time.Duration
+	var err error
+	ttlRaw, ok := data.GetOk("ttl")
+	if !ok {
+		ttl = b.System().DefaultLeaseTTL()
```

Comment:
Don't set this to the default lease TTL if it wasn't set in the request -- it defeats the point of using GetOk! The nicest thing would be the same comment as the postgres/mysql changes -- see if there's an existing value set, and if so use that. Otherwise, set it to 0 so that on credential create you get the correct default values.